tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,335 @@
+    my $vnet_nated = <<'EOF';
+<network>
+  <name>vnet_nated</name>
+  <bridge name="virbr0"/>
+  <forward mode="nat"/>
+  <ip address="192.168.122.1" netmask="255.255.255.0">
+    <dhcp>
+      <range start="192.168.122.2" end="192.168.122.254"/>
+    </dhcp>
+  </ip>
+</network>
+EOF
+    die "There was not the default(NAT BASED NETWORK) virtual network" if (script_run('virsh net-list --all | grep default') != 0);
+
+    #Create NAT BASED NETWORK
+    assert_script_run("virsh net-info default");
+    assert_script_run("virsh net-dumpxml default |tee libvirt_default.xml");
+    assert_script_run("virsh net-undefine default");
+    assert_script_run("echo '$vnet_nated' >> vnet_nated.xml");
+    assert_script_run("virsh net-create vnet_nated.xml");
+    save_screenshot;
+    upload_logs "vnet_nated.xml";
+    assert_script_run("rm -rf vnet_nated.xml");
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "NAT BASED NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_nated --live", 60);
+        script_retry "journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 3, timeout => 60;
+        my $gi_vnet_nated = script_output("journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
+        #Test NAT BASED NETWORK
+        assert_script_run("ssh root\@$gi_vnet_nated 'rcnetwork restart'",           60);
+        assert_script_run("ssh root\@$gi_vnet_nated 'ping -c2 -W1 openqa.suse.de'", 60);
+        save_screenshot;
+        assert_script_run("virsh detach-interface $guest network --current");
+    }
+    #Destroy NATed Network
+    assert_script_run("virsh net-destroy vnet_nated");
+    save_screenshot;
+
+    #Restore default(NATed Network)
+    my $default_path = "/root/libvirt_default.xml";
+    if (script_run("[[ -f $default_path ]]") == 0) { assert_script_run("virsh net-define $default_path", 60); assert_script_run("rm -rf $default_path"); }
+
+##ROUTED NETWORK
+##Prepare Test
+    my $vnet_routed = <<'EOF';
+<network>
+ <name>vnet_routed</name>
+ <bridge name="virbr1"/>
+ <forward mode="route"/>
+ <ip address="192.168.129.1" netmask="255.255.255.0">                                                   <dhcp>
+      <range start="192.168.129.2" end="192.168.129.254" />                                             </dhcp>
+ </ip>
+</network>
+EOF
+
+    #Create and Start ROUTED NETWORK
+    assert_script_run("echo '$vnet_routed' >> vnet_routed.xml");
+    assert_script_run("virsh net-create vnet_routed.xml");
+    save_screenshot;
+    upload_logs "vnet_routed.xml";
+
+    foreach my $guest (keys %xen::guests) {
+        record_info "$guest", "ROUTED NETWORK for $guest";
+        assert_script_run("virsh attach-interface $guest network vnet_routed --live");
+        script_retry "journalctl | grep dnsmasq-dhcp| tail -1| grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 60, retry => 3, timeout => 60;
```

Comment:
just curious, why is `tail -1` needed?

Reply:
could you try `journalctl -t dnsmasq-dhcp` ?

Reply:
I would recommend to add ` --no-pager` option to `journalctl`.

Reply:
Great, working on here now.

Reply:
Finish to add `--no-pager` option to `journalctl` command